isis/tests/FunctionalTestsTopds4.cpp

```diff
@@ -0,0 +1,85 @@
+#include <iostream>
+#include <time.h>
+
+#include <QRegExp>
+#include <QString>
+
+#include "Fixtures.h"
+#include "md5wrapper.h"
+
+#include "topds4.h"
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/topds4.xml").expanded();
+
+TEST_F(SmallCube, FunctionalTestTopds4CurrentTime) {
+  QString templateFile = tempDir.path()+"/current_time.tpl";
+  QString renderedFile = tempDir.path()+"/current_time.txt";
+  std::ofstream of;
+  of.open(templateFile.toStdString());
+  of << "{{currentTime()}}";
+  of.close();
+  QVector<QString> args = {"template=" + templateFile, "to=" + renderedFile};
+  UserInterface options(APP_XML, args);
+
+  topds4(testCube, options);
+
+  std::ifstream renderedStream;
+  renderedStream.open(renderedFile.toStdString());
+  std::string line;
+  std::getline(renderedStream, line);
+  time_t startTime = time(NULL);
+  struct tm *tmbuf = gmtime(&startTime);
+  char yearstr[80];
+  // Just check that the year is correct
+  strftime(yearstr, 80, "%Y", tmbuf);
+  EXPECT_EQ(yearstr, line.substr(0, 4));
+  // Check that the rest is the correct format
+  // YYYY-MM-DDTHH:MM:SS
+  QRegExp timeRegex("\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}");
+  EXPECT_TRUE(QString::fromStdString(line).contains(timeRegex)) << "String [" << line << "] "
+                                                                << "does not match the time format "
+                                                                << "[YYYY-MM-DDTHH:MM:SS].";
```

Comment:
Here's an example output for a failure, this one I had the ouput file contain the MD5 hash,

```
1729: /Users/jmapel/ISIS3/isis/tests/FunctionalTestsTopds4.cpp:43: Failure
1729: Value of: QString::fromStdString(line).contains(timeRegex)
1729:   Actual: false
1729: Expected: true
1729: String [7ca9d00ad2b259d3c4bf95b6e5b477fd] does not match the time format [YYYY-MM-DDTHH:MM:SS].
```